langchain4j-agentic/src/main/java/dev/langchain4j/agentic/internal/AgentExecutor.java

```diff
@@ -0,0 +1,23 @@
+package dev.langchain4j.agentic.internal;
+
+import dev.langchain4j.agentic.cognisphere.DefaultCognisphere;
+
+public record AgentExecutor(AgentInvoker agentInvoker, Object agent) {
+
+    public String agentName() {
+        return agentInvoker.name();
+    }
+
+    public Object invoke(DefaultCognisphere cognisphere) {
```

Comment:
```suggestion
    public Object execute(DefaultCognisphere cognisphere) {
```

To keep it in sync with the name of the interface and not confuse with invoker.

Although executor/invoker are very close names and it is not very clear what is the distinction, and it seems like this method could be as well a part of `AgentInvoker`

Reply:
Method renamed. An `AgentExecutor` is a `record` containing an `AgentInvoker` plus the agent that has to be invoked. If you have a better name, please feel free to suggest it, as you've seen I suck at finding meaningful and understandable names :)